src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,146 @@
+        String userInput = input.nextLine();
+        String userCommand = userInput.split("\\s", 2)[0];
+        int stringLen = userInput.length();
+
+        //adds a todo item to the list
+        if (userCommand.equals("todo")) {
+            try {
+                String taskDetails = userInput.split("\\s", 2)[1];
+                Task newTask = new Todo(taskDetails);
+                taskList.add(newTask);
+                printNewTask(newTask, taskList.size());
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeCommandMissingDescriptionException();
+            }
+        }
+
+        //adds a deadline event to the list
+        else if (userCommand.equals("deadline")) {
+            if (!userInput.contains(" /by ")) {
+                throw new DukeDeadlineMissingByException();
+            }
+            try {
+                String taskDetails = userInput.split("\\s", 2)[1];
+                Task newTask = new Deadline(obtainDeadlineDescription(taskDetails), obtainDeadlineDate(taskDetails));
+                taskList.add(newTask);
+                printNewTask(newTask, taskList.size());
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeCommandMissingDescriptionException();
+            }
+        }
+
+        //adds an event item to the list
+        else if (userCommand.equals("event")) {
+            if (!userInput.contains(" /at ")) {
+                throw new DukeEventMissingAtException();
+            }
+            try {
+                String taskDetails = userInput.split("\\s", 2)[1];
+                Task newTask = new Event(obtainEventDescription(taskDetails), obtainEventDate(taskDetails));
+                taskList.add(newTask);
+                printNewTask(newTask, taskList.size());
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeCommandMissingDescriptionException();
+            }
+        }
+
+        //lists out respective tasks
+        else if (userInput.equals("list")) {
+            if (taskList.isEmpty()) {
+                throw new DukeTaskListEmptyException();
+            }
+            printTaskList(taskList);
+        }
+
+        //marks a task as done
+        else if (userCommand.equals("done")) {
+            try {
+                int i = Integer.parseInt(userInput.substring(5, stringLen)) - 1;
+                taskList.get(i).setDone(true);
+                printCompletedTask(taskList.get(i));
+            } catch (IndexOutOfBoundsException e) {
+                throw new DukeInvalidIndexException();
+            }
+        }
+
+        else if (userCommand.equals("bye")) {
+            System.out.println("Bye, see you soon!");
+            System.exit(0);
+        }
+
+        //handles the case for when an invalid command is entered
+        else {
+            throw new DukeInvalidCommandException();
+        }
+
```

Comment:
You can use enum checks instead of string checks so that it is easier to change the commands in the future.

Or better yet, you can consider using polymorphism so that you can avoid using a bloated if-statement

Reply:
How would I use polymorphism to improve the if-statement?